### PR TITLE
Add a script to check the workspace has private set explicitly

### LINKED ISSUE #452: private: false
TODO: explain this

### PR DESCRIPTION
**Pull request by chee**
this seems better than it being possible to publish something by mistake by not
setting `private: true` amirite?

closes #452

### CODE REVIEW

**Comment by JakeChampion** [comment]
> this seems better than it being possible to publish something by mistake by not setting `private: true` amirite?

You are right ☺️

File: .github/workflows/check-privacy.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    - uses: volta-cli/action@v1
+    - uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
+    - run: npm ci
+    - run: npm run regenerate
+    - run: ./scripts/check-for-changes.bash
```
Comment (JakeChampion) [inline .github/workflows/check-privacy.yml:22]: I think this is meant to call ./scripts/check-privacy.js

Comment (chee) [reply .github/workflows/check-privacy.yml:22]: lol wtf. did i not p[ush ?

File: scripts/check-privacy.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+let message = `
+make sure private is true for packages that should be published, and false for
+those that shouldn't.
+`
+
+for (let cwd of await workspaces.paths()) {
+	let pkg = await readPackage({cwd})
+	if (typeof pkg.private != "boolean") {
+		log(`${cwd}/package.json#private is not a boolean`)
+		code = 130
```
Comment (JakeChampion) [inline scripts/check-privacy.js:20]: Nice

File: scripts/check-privacy.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import {readPackage} from "read-pkg"
+
+let {exit} = process
+let {log} = console
+
+/** exit code */
+let code = 0
+
+let message = `
+make sure private is true for packages that should be published, and false for
```
Comment (JakeChampion) [inline scripts/check-privacy.js:12]: I think this sentence might be the wrong way around

Comment (chee) [reply scripts/check-privacy.js:12]: oh you're right

Comment (chee) [inline scripts/check-privacy.js:12]:
```suggestion
make sure private is false for packages that should be published, and true for
```

**Review by JakeChampion — APPROVED** [review]

**Pull request merged**